### PR TITLE
Add migrations for support-refund

### PR DESCRIPTION
**Pull request by kadewu**
Based on https://github.com/saleor/saleor/pull/12236
Migrations for the https://github.com/saleor/saleor/pull/10350

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Review by korycins — APPROVED** [review]

File: saleor/order/migrations/tasks/saleor3_13.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+@app.task
+def order_eventes_rename_transaction_void_events():
+    qs = OrderEvent.objects.filter(type="transaction_void_requested").order_by("pk")
+    ids = qs.values_list("pk", flat=True)[:BATCH_SIZE]
+    if ids:
+        OrderEvent.objects.filter(pk__in=ids).update(
+            type="transaction_cancel_requested"
+        )
+
+        del ids
```
Comment (IKarbowiak) [inline saleor/order/migrations/tasks/saleor3_13.py:17]: Why do you need to `del ids`?

File: saleor/order/models.py [hunk 2]
```diff
@@ -792,3 +792,37 @@ class Meta:
...
+        blank=True,
+        null=True,
+        on_delete=models.SET_NULL,
+        related_name="+",
+    )
+    app = models.ForeignKey(
+        App, related_name="+", on_delete=models.SET_NULL, null=True, blank=True
+    )
+    order = models.ForeignKey(
+        Order, related_name="granted_refunds", on_delete=models.CASCADE
```
Comment (IKarbowiak) [inline saleor/order/models.py:824]: I'm wondering if it shouldn't be `null=True`?

Comment (korycins) [reply saleor/order/models.py:824]: granted refund is a new feature, so I don't think that we could worry about a case, when order is null. Or maybe you mean some specific case?

Comment (IKarbowiak) [reply saleor/order/models.py:824]: Oh, yes, that's right!

File: saleor/order/migrations/tasks/saleor3_13.py (hunk 1 above)
Comment (kadewu) [reply saleor/order/migrations/tasks/saleor3_13.py:17]: I'm afraid of memory leaks here. I believe that celery did tend to have some issues regarding that, so to be fair, I made that preventively

**Review by IKarbowiak — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

File: saleor/order/migrations/0163_order_events_rename_transaction_events.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    order_eventes_rename_transaction_void_events_task,
+)
+from ... import __version__
+
+BATCH_SIZE = 1000
+
+
+def rename_order_events(apps, _schema_editor):
+    def on_migrations_complete(sender=None, **kwargs):
+        order_eventes_rename_transaction_void_events_task.delay()
```
Comment (maarcingebala) [inline saleor/order/migrations/0163_order_events_rename_transaction_events.py:18]: Typo "eventes" -> "events" here and in other functions as well.

File: saleor/payment/migrations/0039_transactionevent_currency.py [hunk 4]
```diff
@@ -1,25 +1,25 @@
...
-        post_migrate.connect(on_migrations_complete, weak=False, sender=sender)
-    else:
-        TransactionItem = apps.get_model("payment", "TransactionItem")
-        TransactionEvent = apps.get_model("payment", "TransactionEvent")
-        set_default_currency_for_transaction_event(TransactionItem, TransactionEvent)
+    TransactionItem = apps.get_model("payment", "TransactionItem")
+    TransactionEvent = apps.get_model("payment", "TransactionEvent")
+
+    transaction_item = TransactionItem.objects.filter(
+        pk=OuterRef("transaction")
```
Comment (IKarbowiak) [inline saleor/payment/migrations/0039_transactionevent_currency.py:14]: Shouldn't be `pk=OuterRef("transaction_id")`?

File: saleor/payment/migrations/0040_migrate_renamed_fields.py [hunk 5]
```diff
@@ -1,41 +1,74 @@
...
+def transaction_event_migrate_reference_to_psp_reference(qs: QuerySet):
+    qs.update(psp_reference=F("reference"))
+
+
+def transaction_event_migrate_reference_to_psp_reference_task(transaction_event):
+    qs = transaction_event.objects.filter(~Q(psp_reference=F("reference"))).order_by(
+        "-pk"
+    )
+
+    transaction_event_migrate_reference_to_psp_reference(qs)
```
Comment (IKarbowiak) [inline saleor/payment/migrations/0040_migrate_renamed_fields.py:61]: It seems like we do not need the helper right now, and the body of `transaction_event_migrate_reference_to_psp_reference` can be moved here.

Comment (IKarbowiak) [reply saleor/payment/migrations/0040_migrate_renamed_fields.py:61]: Same for all other methods that are called in `migrate_data_for_renamed_fields`

File: saleor/payment/migrations/0041_add_calculation_transaction_events.py [hunk 6]
```diff
@@ -1,37 +1,157 @@
...
-from ... import __version__
 
 
-def create_transaction_events(apps, schema_editor):
-    def on_migrations_complete(sender=None, **kwargs):
-        create_event_for_authorized_task.delay()
-        create_event_for_canceled_task.delay()
-        create_event_for_charged_task.delay()
-        create_event_for_refunded_task.delay()
+def create_transaction_events(transaction_event, transactions_qs, type):
```
Comment (IKarbowiak) [inline saleor/payment/migrations/0041_add_calculation_transaction_events.py:16]: Is it safe to not use batches here?

File: saleor/payment/migrations/0039_transactionevent_currency.py (hunk 4 above)
Comment (kadewu) [reply saleor/payment/migrations/0039_transactionevent_currency.py:14]: It's equivalent.

File: saleor/payment/migrations/0040_migrate_renamed_fields.py (hunk 5 above)
Comment (kadewu) [reply saleor/payment/migrations/0040_migrate_renamed_fields.py:61]: The main idea of leaving those functions is to make it easier to see what logic was moved from celery to migration.

File: saleor/payment/migrations/0041_add_calculation_transaction_events.py (hunk 6 above)
Comment (kadewu) [reply saleor/payment/migrations/0041_add_calculation_transaction_events.py:16]: Yes, as transactions are in preview feature.

**Pull request merged**